core/services/chainlink/application.go

```diff
@@ -532,39 +548,23 @@ func (app *ChainlinkApplication) stop() error {
 		// Stop services in the reverse order from which they were started
 
 		logger.Debug("Stopping Scheduler...")
-		app.Scheduler.Stop()
+		merr = multierr.Append(merr, app.Scheduler.Stop())
 
 		logger.Debug("Stopping HeadTracker...")
 		merr = multierr.Append(merr, app.HeadTracker.Stop())
 
 		for i := len(app.subservices) - 1; i >= 0; i-- {
 			service := app.subservices[i]
-			logger.Debugw(fmt.Sprintf("Closing service %v...", i), "serviceType", reflect.TypeOf(service))
+			logger.Debugw("Closing service...", "serviceType", reflect.TypeOf(service))
 			merr = multierr.Append(merr, service.Close())
 		}
 
-		logger.Debug("Stopping Scheduler...")
-		app.Scheduler.Stop()
-		logger.Debug("Stopping balanceMonitor...")
-		merr = multierr.Append(merr, app.balanceMonitor.Stop())
-		logger.Debug("Stopping JobSubscriber...")
-		merr = multierr.Append(merr, app.JobSubscriber.Stop())
-		logger.Debug("Stopping FluxMonitor...")
-		app.FluxMonitor.Stop()
-		logger.Debug("Stopping EventBroadcaster...")
-		merr = multierr.Append(merr, app.EventBroadcaster.Stop())
-		logger.Debug("Stopping LogBroadcaster...")
-		merr = multierr.Append(merr, app.LogBroadcaster.Stop())
-		logger.Debug("Stopping RunQueue...")
-		app.RunQueue.Stop()
-		logger.Debug("Stopping StatsPusher...")
-		merr = multierr.Append(merr, app.StatsPusher.Close())
-		logger.Debug("Stopping explorerClient...")
-		merr = multierr.Append(merr, app.explorerClient.Close())
```

Comment:
Has the order been preserved?

Reply:
Almost but not quite: It follows the order of declarations inside `NewApplication`, and I moved RunQueue earlier to start with the RunManager that doesn't fit into the Service interface. `jobSubscriber` and `balanceMonitor` now fit into subservices, so they were added.

| Old                            | New                                |
|--------------------------------|------------------------------------|
| store                          | store                              |
|                                | RunManager.ResumeAllInProgress |
| explorerClient                 | explorerClient |
| StatsPusher                    | StatsPusher |
| RunQueue                       | |
| RunManager.ResumeAllInProgress |  |
|                                | gasUpdater |
|                                | databaseBackup |
|                                | RunQueue |
| LogBroadcaster                 | LogBroadcaster |
| EventBroadcaster               | EventBroadcaster |
| FluxMonitor                    | FluxMonitor |
| gasUpdater                     | |
| databaseBackup                 | |
|                                | jobSubscriber |
|                                | balanceMonitor |
| promReporter                   | promReporter |
| concretePW                     | concretePw |
| jobSpawner                     | jobSpawner |
| pipelineRunnner                | pipelineRunner |
| bptxm                          | bptxm |
| headBroadcaster                | headBroadcaster |